src/ResourceManager/Sql/Commands.Sql/Instance Failover Group/Cmdlet/NewAzureSqlInstanceFailoverGroup.cs

```diff
@@ -0,0 +1,189 @@
+
+        /// <summary>
+        /// Gets or sets the failover policy without data loss for the Sql Azure Instance Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The failover policy of the Azure SQL Database Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        [PSDefaultValue(Help = "Automatic")]
+        public FailoverPolicy FailoverPolicy { get; set; }
+
+        /// <summary>
+        /// Gets or sets the grace period with data loss for the Sql Azure Instance Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Interval before automatic failover is initiated if an outage occurs on the primary server and failover cannot be completed without data loss.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateRange(1, int.MaxValue/60)]
+        [PSDefaultValue(Help = "1")]
+        public int GracePeriodWithDataLossHours { get; set; }
+
+        /// <summary>
+        /// Gets or sets the failover policy for read only endpoint of the Sql Azure Instance Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Whether an outage on the secondary server should trigger automatic failover of the read-only endpoint. This feature is not yet supported.")]
+        [ValidateNotNullOrEmpty]
+        public AllowReadOnlyFailoverToPrimary AllowReadOnlyFailoverToPrimary { get; set; }
+
+        /// <summary>
+        /// Get the entities from the service
+        /// </summary>
+        /// <returns>The list of entities</returns>
+        protected override IEnumerable<AzureSqlInstanceFailoverGroupModel> GetEntity()
+        {
+            // We try to get the failover group.  Since this is a create, we don't want the failover group to exist
+            try
+            {
+                ModelAdapter.GetInstanceFailoverGroup(this.ResourceGroupName, this.Location, this.Name);
+            }
+            catch
+            {
+                return null; 
+            }
+
+            // The Failover Group already exists
+            throw new PSArgumentException(string.Format(Properties.Resources.FailoverGroupNameExists, this.Name, this.Location), "FailoverGroupName");
+        }
+
+        /// <summary>
+        /// Create the model from user input
+        /// </summary>
+        /// <param name="model">Model retrieved from service</param>
+        /// <returns>The model that was passed in</returns>
+        protected override IEnumerable<AzureSqlInstanceFailoverGroupModel> ApplyUserInputToModel(IEnumerable<AzureSqlInstanceFailoverGroupModel> model)
+        {
+            string location = Location;
+            List<AzureSqlInstanceFailoverGroupModel> newEntity = new List<AzureSqlInstanceFailoverGroupModel>();
+            object parameterValue;
+
+            int? gracePeriod = null;
+            if (!FailoverPolicy.ToString().Equals("Manual"))
```

Comment:
@jugeorge we should check if `FailoverPolicy` is provided before accessing any properties

Reply:
Because there is a default value for FailoverPolicy (it gets set to Automatic when nothing has been provided), doesn't that mean the value will always exist?